src/elf.c

```diff
@@ -320,6 +320,7 @@ bool elf_open(elf_t *e, const char *input)
     if (e->raw_data == MAP_FAILED) {
         release(e);
         free(path);
+        close(fd);
```

Comment:
Follow the method used by `cache_create` for employing `goto` to release resources, which results in more concise code.